services/tables/src/main/java/com/linkedin/openhouse/tables/utils/TableUUIDGenerator.java

```diff
@@ -64,12 +66,44 @@ public UUID generateUUID(CreateUpdateTableRequestBody createUpdateTableRequestBo
   public UUID generateUUID(IcebergSnapshotsRequestBody icebergSnapshotsRequestBody) {
     return extractUUIDFromSnapshotJson(
             icebergSnapshotsRequestBody.getJsonSnapshots(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getDatabaseId(),
-            icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableId())
+            extractFromTblPropsIfExists(
+                getTableURI(icebergSnapshotsRequestBody),
+                icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableProperties(),
+                DB_RAW_KEY),
+            extractFromTblPropsIfExists(
+                getTableURI(icebergSnapshotsRequestBody),
+                icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody().getTableProperties(),
+                TBL_RAW_KEY))
         .orElseGet(
             () -> generateUUID(icebergSnapshotsRequestBody.getCreateUpdateTableRequestBody()));
   }
 
+  /** Simple helper method to obtain tableURI from requestBody. */
+  private String getTableURI(IcebergSnapshotsRequestBody icebergSnapshotsRequestBody) {
```

Comment:
the purpose of tableURI seem to just be error log, could we retrieve it from the properties instead? and make it implementation friendly?

Reply:
Yeah, better to formulate tableURI in method itself rather than passing as param.

Reply:
Yes it was for logging. It cannot be retrieved from properties as the logging is exactly for the case where properties is missing these info (API has mandate table Id and dbId, so there must be `tableId` and `dbId` presented at the API/TableDTO layer) 

Also, @HotSushi  what does it mean by "make it implementation friendly"? 
@rohitkum2506  what's the concern of passing method's result as a param? Didn't get that.

Reply:
I meant not needing `getTableURI` as a separate method

Reply:
^ this is not needed anymore right?

Reply:
yes I think I removed it from the newest version.